child_compassion/models/compassion_hold.py

```diff
@@ -420,6 +420,13 @@ def postpone_no_money_cron(self):
     def beneficiary_hold_removal(self, commkit_data):
         data = commkit_data.get("BeneficiaryHoldRemovalNotification")
         hold = self.search([("hold_id", "=", data.get("HoldID"))])
+
+        # avoid realising a hold (and related child) that has already been released
+        if hold and hold.state == "expired":
+            logger.warning("Received Beneficiary Hold Removal order from GMC for already expired hold.")
+            # return empty list because no hold were created or modified
+            return []
```

Comment:
I think returning the hold id would actually let us open the hold from the message which can be handy. It doesn't change anything else on the logic I think.
```suggestion
            return hold.ids
```

Reply:
I didn't knew about this feature so yes good suggestion. 
I will have to add a commit on compassion-switzerland because if the super() call return some hold it might create the unplanned exit communication (that's why I returned an empty list)